tmp/build/generate-manifests.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# to generate Maistra OLM metadata: DEPLOYMENT_FILE=deploy/maistra-operator.yaml MANIFESTS_DIR=manifests-maistra tmp/build/generate-manifests.sh
+# to generate ServiceMesh OLM metadata: DEPLOYMENT_FILE=deploy/servicemesh-operator.yaml MANIFESTS_DIR=manifests-servicemesh tmp/build/generate-manifests.sh
+
+set -e
+
+: ${DEPLOYMENT_FILE:?"Need to set location of the source deployment file, e.g. deploy/maistra-operator.yaml"}
+: ${MANIFESTS_DIR:?"Need to set location of the manifests directory, e.g. manifests/"}
+: ${MAISTRA_VERSION:?"Need to set maistra version, e.g. 1.0.0"}
+
+BUNDLE_DIR=${MANIFESTS_DIR}/${MAISTRA_VERSION}
+MY_LOCATION="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ ! -d "$BUNDLE_DIR" ]; then
+  mkdir -p "$BUNDLE_DIR"
+fi
+
+function generateControlPlanesCrd() {
+  yq -s -y '.[] | select(.kind=="CustomResourceDefinition" and .metadata.name=="controlplanes.istio.openshift.com") | .' ${DEPLOYMENT_FILE} > ${BUNDLE_DIR}/controlplanes.crd.yaml
+}
+
+function generateServiceMeshControlPlanesCrd() {
+  yq -s -y '.[] | select(.kind=="CustomResourceDefinition" and .metadata.name=="servicemeshcontrolplanes.maistra.io") | .' ${DEPLOYMENT_FILE} > ${BUNDLE_DIR}/servicemeshcontrolplanes.crd.yaml
+}
+
+function generateServiceMeshMemberRollsCrd() {
+  yq -s -y '.[] | select(.kind=="CustomResourceDefinition" and .metadata.name=="servicemeshmemberrolls.maistra.io") | .' ${DEPLOYMENT_FILE} > ${BUNDLE_DIR}/servicemeshmemberrolls.crd.yaml
+}
+
+function generateCSV() {
+  IMAGE_SRC=$(yq -s -r '.[] | select(.kind=="Deployment" and .metadata.name=="istio-operator") | .spec.template.spec.containers[0].image' ${DEPLOYMENT_FILE})
+  if [ "$IMAGE_SRC" == "" ]; then
+     echo "generateCSV(): Operator image source is empty, please verify source yaml/path to the field."
+     exit 1
+  fi
+
+  CLUSTER_ROLE_RULES=$(yq -s -y '.[] | select(.kind=="ClusterRole" and .metadata.name=="istio-operator") | .rules' ${DEPLOYMENT_FILE} | sed 's/^/        /')
+  if [ "$CLUSTER_ROLE_RULES" == "null" ]; then
+     echo "generateCSV(): istio-operator cluster role source is empty, please verify source yaml/path to the field."
+     exit 1
+  fi
+
+  CSV_PATH=${BUNDLE_DIR}/maistra.v${MAISTRA_VERSION}.clusterserviceversion.yaml
+  cp ${MY_LOCATION}/manifest-templates/clusterserviceversion.yaml ${CSV_PATH}
+  sed -i -e 's/__VERSION__/'${MAISTRA_VERSION}'/g' ${CSV_PATH}
+  sed -i -e 's+__IMAGE_SRC__+'${IMAGE_SRC}'+g' ${CSV_PATH}
+  sed -i -e '/__CLUSTER_ROLE_RULES__/{
+    s/__CLUSTER_ROLE_RULES__//
+    r '<(echo "$CLUSTER_ROLE_RULES")'
+  }' ${CSV_PATH}
+}
+
+function generatePackage() {
+  PACKAGE_PATH=${MANIFESTS_DIR}/maistra.package.yaml
```

Comment:
You can use `local` for local variables like this (and have the local variable name in lower case)

Reply:
Fixed